expression/constant_fold.go

```diff
@@ -86,18 +86,49 @@ func foldConstant(expr Expression) (Expression, bool) {
 		}
 
 		args := x.GetArgs()
-		canFold := true
+		sc := x.GetCtx().GetSessionVars().StmtCtx
+		argIsConst := make([]bool, len(args))
+		hasNullArg := false
+		allConstArg := true
 		isDeferredConst := false
 		for i := 0; i < len(args); i++ {
 			foldedArg, isDeferred := foldConstant(args[i])
 			x.GetArgs()[i] = foldedArg
-			_, conOK := foldedArg.(*Constant)
-			if !conOK {
-				canFold = false
+			con, conOK := foldedArg.(*Constant)
+			if conOK {
```

Comment:
how about:
```go
argIsConst[i] = conOk
allConstArg = allConstArg && conOk
hasNullArg = hasNullArg || (conOk && con.Value.IsNull())
```

Maybe this is more clearer.

Reply:
thanks, patch updated